internal/scan-manager/core/src/test/java/org/wso2/security/tools/scanmanager/core/service/ServiceTestDataProvider.java

```diff
@@ -0,0 +1,162 @@
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import org.testng.annotations.DataProvider;
+import org.wso2.security.tools.scanmanager.common.external.model.Log;
+import org.wso2.security.tools.scanmanager.common.external.model.Scan;
+import org.wso2.security.tools.scanmanager.common.external.model.ScanFile;
+import org.wso2.security.tools.scanmanager.common.external.model.ScanProperty;
+import org.wso2.security.tools.scanmanager.common.external.model.Scanner;
+import org.wso2.security.tools.scanmanager.common.external.model.ScannerApp;
+import org.wso2.security.tools.scanmanager.common.external.model.ScannerField;
+import org.wso2.security.tools.scanmanager.common.model.LogType;
+import org.wso2.security.tools.scanmanager.common.model.ScanPriority;
+import org.wso2.security.tools.scanmanager.common.model.ScanStatus;
+import org.wso2.security.tools.scanmanager.common.model.ScanType;
+import org.wso2.security.tools.scanmanager.common.model.ScannerType;
+import org.wso2.security.tools.scanmanager.core.config.ScanManagerConfiguration;
+import org.wso2.security.tools.scanmanager.core.model.Container;
+
+import java.io.IOException;
+import java.sql.Timestamp;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static org.wso2.security.tools.scanmanager.core.util.Constants.CONTAINER_APP_LABEL_NAME;
+import static org.wso2.security.tools.scanmanager.core.util.Constants.CONTAINER_ENV_NAME_SCAN_MANAGER_HOST;
+import static org.wso2.security.tools.scanmanager.core.util.Constants.CONTAINER_ENV_NAME_SCAN_MANAGER_PORT;
+import static org.wso2.security.tools.scanmanager.core.util.Constants.CONTAINER_SCANNER_LABEL_NAME;
+import static org.wso2.security.tools.scanmanager.core.util.Constants.CONTAINER_SCAN_JOB_ID_LABEL_NAME;
+import static org.wso2.security.tools.scanmanager.core.util.ScanManagerTestConstants.SCANNER_APP_ID;
+import static org.wso2.security.tools.scanmanager.core.util.ScanManagerTestConstants.SCANNER_APP_NAME;
+import static org.wso2.security.tools.scanmanager.core.util.ScanManagerTestConstants.TEST_PRODUCT_ID;
+import static org.wso2.security.tools.scanmanager.core.util.ScanManagerTestConstants.TEST_SCANNER_ID;
+import static org.wso2.security.tools.scanmanager.core.util.ScanManagerTestConstants.TEST_SCAN_ID;
+
+/**
+ * Base class for service tests.
+ */
+public class ServiceTestDataProvider {
+
+    @DataProvider(name = "getScannerData")
+    public static Object[][] getScannerData() {
+        Scan scan = buildScan();
+        return new Object[][]{{scan.getScanner()}};
+    }
+
+    private static Scan buildScan() {
+        Scan scan = null;
+        Scanner scanner;
+        ScannerApp scannerApp;
+        ScannerField scannerField;
+        ScanFile scanFile;
+        ScanProperty scanProperty;
+
+        scanner = new Scanner();
```

Comment:
It would be neater to to this in the declaration itself (reduce the line count too). Is there a reason for separating the declaration and this assignment?

Same applies for other fields.